### PR TITLE
Feature/splash screen

### PR DESCRIPTION
**Pull request by manetcom**

### CODE REVIEW

File: FHICORC.Core/Services/DecoderServices/CertificateRevocationService.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        public CertificateRevocationService(IRevocationBatchService revocationBatchService)
+        {
+            _revocationBatchService = revocationBatchService;
+            _bloomFilterBuckets = MobileUtils.FillBloomBuckets();
+        }
+
+
+        public async Task<bool> IsCertificateRevoked(ITokenPayload token, string signatureBase64EncodedHash)
+        {
+            if(token is DCCPayload payload)
```
Comment (tslaNetcompany) [inline FHICORC.Core/Services/DecoderServices/CertificateRevocationService.cs]: add space between if statement and ()

File: FHICORC.Core/Services/DecoderServices/CertificateRevocationService.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            StringBuilder Sb = new StringBuilder();
+
+            foreach (Byte b in bytes)
+                Sb.Append(b.ToString("x2"));
+
+            return Sb.ToString();
+        }
+
+
+        public bool CheckHashInRevocationBatchesAsync(IEnumerable<RevocationBatch> revocationBatches, string uciBase64EndodedHash, string countryCodeUciBase64EndodedHash, string signatureBase64EncodedHash, bool isParallel=false)
```
Comment (tslaNetcompany) [inline FHICORC.Core/Services/DecoderServices/CertificateRevocationService.cs]: isParallel = false

File: FHICORC.Core/Services/Interface/IFetchRevocationBatchesFromBackednService.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+﻿using System.Threading.Tasks;
+
+namespace FHICORC.Core.Services.Interface
+{
+    public interface IFetchRevocationBatchesFromBackednService
```
Comment (tslaNetcompany) [inline FHICORC.Core/Services/Interface/IFetchRevocationBatchesFromBackednService.cs]: IFetchRevocationBatchesFromBackendService

File: FHICORC.Core/Data/Models/RevocationBatch.cs [hunk 4]
```diff
@@ -0,0 +1,28 @@
+﻿using FHICORC.Core.Services.Enum;
+using Newtonsoft.Json;
+using SQLite;
+using System;
+
+namespace FHICORC.Core.Data.Models
+{
+    public class RevocationBatch : DatabaseEntityBase
```
Comment (tslaNetcompany) [inline FHICORC.Core/Data/Models/RevocationBatch.cs:8]: add constructor

File: FHICORC.Core/Services/DecoderServices/FetchRevocationBatchesFromBackednService.cs [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+        //    
+        //}
+
+
+
+
+    }
+
+
+    //    public class Monitor
```
Comment (tslaNetcompany) [inline FHICORC.Core/Services/DecoderServices/FetchRevocationBatchesFromBackednService.cs]: Is this needed?

File: FHICORC.Core/Services/DecoderServices/FetchRevocationBatchesFromBackednService.cs [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+        }
+
+        public Task HasChangedTask()
+        {
+            return tcs.Task; 
+        }
+
+
+        //static async void Run()
```
Comment (tslaNetcompany) [inline FHICORC.Core/Services/DecoderServices/FetchRevocationBatchesFromBackednService.cs]: Is this needed?

File: FHICORC.Core/Services/Utils/BloomFilterUtils.cs [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+        }
+
+    }
+
+
+    public class BloomStats
+    {
+        public int ExpectedElements { get; set; }
+        public double ErrorRate { get; set; }
+        public int m { get; set; }
```
Comment (tslaNetcompany) [inline FHICORC.Core/Services/Utils/BloomFilterUtils.cs:135]: PascalCase

File: FHICORC.Core/Services/Utils/MobileUtils.cs [hunk 8]
```diff
@@ -0,0 +1,147 @@
...
+using System.Text;
+using BloomFilter;
+using FHICORC.Core.Data.Models;
+using FHICORC.Core.Services.Model;
+
+namespace FHICORC.Core.Services.Utils
+{
+    public static class MobileUtils
+    {
+        public static bool ContainsCertificateFilterMobile(string uciHash, string countryCodeUciHash, string signatureHash, IEnumerable<RevocationBatch> revocationBatches, List<BucketItem> bloomFilterBuckets, bool isParallel=true)
```
Comment (tslaNetcompany) [inline FHICORC.Core/Services/Utils/MobileUtils.cs]: bool isParallel = true

File: FHICORC/App.xaml.cs [hunk 9]
```diff
@@ -23,10 +25,14 @@ public partial class App : Application
         private readonly IPreferencesService _preferencesService;
         private readonly ITextService _textService;
         private readonly INavigationService _navigationService;
+        private readonly IRevocationBatchService _revocationBatchDataManager;
+        private readonly IRevocationDeleteExpiredBatchService _revocationDeleteExpiredBatchService;
         private IPublicKeyService _publicKeyDataManager;
         private IBusinessRulesService _businessRulesDataManager;
         private IValueSetService _valueSetService;
 
+        private readonly IFetchRevocationBatchesFromBackednService _fetchRevocationBatchesFromBackednService;
```
Comment (tslaNetcompany) [inline FHICORC/App.xaml.cs]: _fetchRevocationBatchesFromBackendService

File: FHICORC/App.xaml.cs [hunk 10]
```diff
@@ -41,6 +47,9 @@ public App()
             _publicKeyDataManager = IoCContainer.Resolve<IPublicKeyService>();
             _businessRulesDataManager = IoCContainer.Resolve<IBusinessRulesService>();
             _valueSetService = IoCContainer.Resolve<IValueSetService>();
+            _revocationBatchDataManager = IoCContainer.Resolve<IRevocationBatchService>();
+            _revocationDeleteExpiredBatchService = IoCContainer.Resolve<IRevocationDeleteExpiredBatchService>();
+            _fetchRevocationBatchesFromBackednService = IoCContainer.Resolve<IFetchRevocationBatchesFromBackednService>();
```
Comment (tslaNetcompany) [inline FHICORC/App.xaml.cs]: _fetchRevocationBatchesFromBackednService

**Review by tslaNetcompany — CHANGES_REQUESTED** [review]

**Review by teddy515 — APPROVED** [review]

**Pull request merged**